src/compas/files/gltf.py

```diff
@@ -0,0 +1,162 @@
+
+        vertex_coordinate_data = {self.root: [0, 0, 0]}
+        vertex_data = {}
+        edge_data = []
+
+        while edge_list:
+            parent, child, attributes = edge_list.pop(0)
+            if parent not in vertex_coordinate_data:
+                edge_list.append((parent, child, attributes))
+                continue
+
+            parent_vector = vertex_coordinate_data[parent]
+            matrix = attributes['matrix']
+
+            vertex_coordinate_data[child] = self._inhomogeneous_transformation(matrix, parent_vector)
+            vertex_data[child] = attributes
+            edge_data.append((parent, child))
+
+        self.scene_data = {
+            'vertex_data': vertex_data,  # {child_name: {'matrix': transformation matrix from parent to child, 'geometry': mesh name, 'time': timestamp}}
+            'vertices': vertex_coordinate_data,  # {vertex_name: [x, y, z]]}
+            'edges': edge_data,  # [(parent_name, child_name])
+        }
+
+    def _inhomogeneous_transformation(self, matrix, vector):
+        transformation_matrix = np.array(matrix)
+        embedded_vector = np.array(self._embed_vector(vector))
+
+        embedded_result = transformation_matrix.dot(embedded_vector)
+
+        return list(self._pullback_vector(embedded_result))
+
+    def _embed_vector(self, vector):
+        return vector + [1.0]
+
+    def _pullback_vector(self, vector):
+        return vector[:3]
+
+
+# ==============================================================================
+# Main
+# ==============================================================================
+
+if __name__ == '__main__':
+
+    import os
+    import compas
+
+    from compas.datastructures import Mesh, Network
+    # from compas_viewers import MultiMeshViewer
+    from compas.utilities import download_file_from_remote
+
+    source_gltf = 'https://raw.githubusercontent.com/KhronosGroup/glTF-Sample-Models/master/2.0/GearboxAssy/glTF/GearboxAssy.gltf'
+    source_bin = 'https://raw.githubusercontent.com/KhronosGroup/glTF-Sample-Models/master/2.0/GearboxAssy/glTF/GearboxAssy0.bin'
+    filepath_gltf = os.path.join(compas.APPDATA, 'data', 'gltfs', 'khronos', 'GearboxAssy.gltf')
+    filepath_bin = os.path.join(compas.APPDATA, 'data', 'gltfs', 'khronos', 'GearboxAssy0.bin')
+
+    download_file_from_remote(source_gltf, filepath_gltf, overwrite=False)
+    download_file_from_remote(source_bin, filepath_bin, overwrite=False)
+
+    gltf = GLTF(filepath_gltf)
+
+    meshes = {
+        name: Mesh.from_vertices_and_faces(data['vertices'], data['faces'])
+        for name, data in gltf.parser.faces_and_vertices_dict.items()
+    }
+
+    for name in meshes:
+        meshes[name].name = name
+
+    scene_tree = Network.from_vertices_and_edges(gltf.parser.scene_data['vertices'], gltf.parser.scene_data['edges'])
+    scene_tree.plot()
+
+    # Usage of MultiMeshViewer unclear
+    # viewer = MultiMeshViewer()
+    # viewer.meshes = meshes.values()
+    # viewer.show()
```

Comment:
?

Reply:
I could clarify with "Usage of MultiMeshViewer unclear _to me_".  This seems to follow the pattern established in stl.py, but when I tried to run stl.py and gltf.py, it didn't work.  I left this in here so that some one might explain viewers to me (but would be removed/fixed before merging).

Reply:
okay, nor problem. do you have `compas_viewers` installed?

Reply:
yes

Reply:
also, in py3, `keys` and `values` are iterators by default so you may want to convert the output to an actual list
```python
viewer.meshes = list(meshes.values())
```

Reply:
https://compas-dev.github.io/examples/mesh-subd-comparison/mesh-subd-comparsion.html

Reply:
It seems that the import statement in stl.py was outdated and should be replaced with 
    `from compas_viewers.meshviewer import MeshViewer`
This is what was causing me confusion.